File: busca-binaria/busca-binaria-01.php
<?php
    $array = [0,1,2,3,4,5,6,7,8,9];
    $int = 1;

    echo buscaBinaria($array, 4);

    function buscaBinaria(array $array, int $valor){
        $tamanhoArray = count($array);
        $baixo = 0;
        $alto =  $tamanhoArray - 1;

        for($i = 0; $i < $tamanhoArray; $i++){
            $meio = round(($alto + $baixo)/2, 0, PHP_ROUND_HALF_DOWN);


            if($meio == $valor){
                return $valor;
            }else if($meio < $valor){
                $baixo = $meio + 1;
            }else if($meio > $valor){
                $alto = $meio - 1;
            }
        }
        return "Não há {$valor}";
    }
?>

File: busca-binaria/busca-binaria-01.py
def buscaBinaria(lista, item):
    indexMaisAlto = len(lista) - 1
    
    baixo = 0
    alto = indexMaisAlto

    while baixo <= alto:
        meio = round((baixo + alto) / 2)
        chute = lista[meio]

        if chute == item:
            return meio
        elif chute < item:
            baixo = meio + 1
        else:
            alto = meio - 1
    return None

lista = [1,3,5,7,9]

print(buscaBinaria(lista, 9))


